### PR TITLE
[8.0][ADD] website_sale_survey_module

### PR DESCRIPTION
**Pull request by angelmoya**
Add website sale survey after payment in online shop

### CODE REVIEW

**Comment by angelmoya** [comment]
About CLA, I already had signed but now I change my username, my email is the same, someone could fix it, my previous username was angelmoya, sorry

**Comment by pedrobaeza** [comment]
CLA checked

File: website_sale_survey/README.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+===================
+Website Sale Survey
+===================
+
+This module extends the functionality of online sale orders to support surveys
+and to allow you to define a default survey and show it to customer after payment
```
Comment (pedrobaeza) [inline website_sale_survey/README.rst]: s/and show/and to show

File: website_sale_survey/README.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+#. Go to /Marketing/Surveys/Surveys, create a survey and set it as default
+
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/website/8.0
```
Comment (pedrobaeza) [inline website_sale_survey/README.rst]: Repository ID, not repository name

File: website_sale_survey/README.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (pedrobaeza) [inline website_sale_survey/README.rst]: Remove this once you have found ID

File: website_sale_survey/__init__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+#
+#    Jamotion GmbH, Your Odoo implementation partner
```
Comment (pedrobaeza) [inline website_sale_survey/__init__.py]: Please use short headers: https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py#L1-L3

File: website_sale_survey/i18n/website_sale_survey.pot [hunk 5]
```diff
@@ -0,0 +1,55 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline website_sale_survey/i18n/website_sale_survey.pot]: Don't include POT files

File: website_sale_survey/README.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+This module extends the functionality of online sale orders to support surveys
+and to allow you to define a default survey and show it to customer after payment
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+#. Go to /Marketing/Surveys/Surveys, create a survey and set it as default
```
Comment (pedrobaeza) [inline website_sale_survey/README.rst]: Default? Can it be more specific like "Use it in e-commerce"?

**Comment by angelmoya** [comment]
Thanks @pedrobaeza all comments are fixed.

**Comment by rafaelbn** [comment]
Hi, I tested and I got 500: Internal Server Error

```

500: Internal Server Error
Error

Error message:

Element '<div class="oe_structure" data-oe-id="1336" data-oe-source-id="1291" data-oe-xpath="/data/div" data-oe-model="ir.ui.view" data-oe-field="arch">' cannot be located in parent view

Error context:
View `confirmation_survey`
[view_id: 1336, xml_id: website_sale_survey.confirmation_survey, model: n/a, parent_id: 1291]

Traceback

Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/website/models/ir_http.py", line 199, in _handle_exception
    response = super(ir_http, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_http.py", line 145, in _handle_exception
    return request._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/http.py", line 673, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/http.py", line 1377, in get_response
    result.flatten()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/http.py", line 1248, in flatten
    self.response.append(self.render())
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/http.py", line 1242, in render
    context=request.context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/website_event_register_free/model/ir_ui_view.py", line 35, in render
    context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/website/models/ir_ui_view.py", line 173, in render
    return super(view, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_ui_view.py", line 1024, in render
    return self.pool[engine].render(cr, uid, id_or_xml_id, qcontext, loader=loader, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_qweb.py", line 263, in render
    return self.render_node(self.get_template(id_or_xml_id, qwebcontext), qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_qweb.py", line 192, in get_template
    xml_doc = qwebcontext.loader(name)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_ui_view.py", line 1022, in loader
    return self.read_template(cr, uid, name, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in read_template
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/tools/cache.py", line 122, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_ui_view.py", line 854, in read_template
    arch = self.read_combined(cr, uid, view_id, fields=['arch'], context=context)['arch']
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_ui_view.py", line 551, in read_combined
    cr, uid, arch_tree, root_id, base.model, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_ui_view.py", line 499, in apply_view_inheritance
    source = self.apply_inheritance_specs(cr, uid, source, specs_tree, view_id, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_ui_view.py", line 477, in apply_inheritance_specs
    self.raise_view_error(cr, uid, _("Element '%s' cannot be located in parent view") % tag, inherit_id, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3153560-249-95ee13/openerp/addons/base/ir/ir_ui_view.py", line 362, in raise_view_error
    raise AttributeError(message)
AttributeError: Element '<div class="oe_structure" data-oe-id="1336" data-oe-source-id="1291" data-oe-xpath="/data/div" data-oe-model="ir.ui.view" data-oe-field="arch">' cannot be located in parent view

Error context:
View `confirmation_survey`
[view_id: 1336, xml_id: website_sale_survey.confirmation_survey, model: n/a, parent_id: 1291]

```

**Comment by angelmoya** [comment]
@rafaelbn Maybe it isn't compatible with other module of this branch, I tested another time on a clean installation and works fine, I do "force rebuild" on runbot to test on it.

**Comment by angelmoya** [comment]
@rafaelbn Checked on runbot and is working fine, I did this:
- Enter on shop without login, buy an item, process payment, all ok, no survey configured then no survey showed.
- Then I login as admin, config survey as default and logout.
- Enter on shop without login, buy an item, process payment, all ok, on process OK page then a button is showed to fill survey.

Could you reproduce las error? steps to do?

**Review by rafaelbn — CHANGES_REQUESTED** [review]
Hi @angelmoya , the error said in https://github.com/OCA/website/pull/249#issuecomment-246095215 comes when you buy (after payment) logged in Odoo.

**Comment by angelmoya** [comment]
@rafaelbn It's fixed, I test it on my server, but runbot is not working to test

**Review by rafaelbn — APPROVED** [review]

**Comment by rafaelbn** [comment]
I would like to test it in runbot when available :-)

**Review by susport — COMMENTED** [review]
👍

**Pull request merged**